src/rgw/rgw_rest_pubsub.cc

```diff
@@ -626,6 +640,34 @@ class RGWPSDeleteNotif_ObjStore_S3 : public RGWPSDeleteNotifOp {
     }
   }
 
+  void delete_all_notifications(const rgw_pubsub_bucket_topics& bucket_topics, const RGWPubSub::BucketRef& b, optional_yield y) {
```

Comment:
* please change it to be a standalone function. pass as arguments all data members that are used inside it
* please don't pass `op_ret`. instead return an int from the function and assign to `op_ret` in the calling functions